src/ext/service.rs

```diff
@@ -99,65 +98,69 @@ impl ServiceExt for dc::Service {
                        sources: &'b sources::Sources)
                        -> Result<Sources<'b>> {
         // Get our `context`, if any.
-        let source_mount_dir = self.source_mount_dir()?;
-        let repo_subdir = self.repo_subdir()?;
-        let context = self.context()?.map(|ctx| (source_mount_dir, repo_subdir, ctx.clone()));
+        let container_path = self.source_mount_dir()?;
+        let source_subdirectory = self.source_subdirectory()?;
+
+        let context = self.context()?
+            .and_then(|ctx| {
+                // human_alias is called on every context when constructing Sources
+                let alias = &ctx.human_alias().expect("human_alias failed on a context that worked previously");
+                sources.find_by_alias(alias)
+            })
+            .and_then(|source| {
+                Some(SourceMount { container_path, source, source_subdirectory })
+            });
```

Comment:
The `find_by_alias` above has moved up here, instead of being down in the iterator implementation.